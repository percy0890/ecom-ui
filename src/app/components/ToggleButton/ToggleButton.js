import React from 'react';
import PropTypes from 'prop-types';
import './ToggleButton.scss';

export default function ToggleButton({ customClass = '', onChange, checked }) {
  return (
    <label className={`switch ${customClass}`}>
      <input type="checkbox" checked={checked} onChange={onChange} />
      <span className="slider round" />
    </label>
  );
}

ToggleButton.propTypes = {
  customClass: PropTypes.string,
  onChange: PropTypes.func,
  checked: PropTypes.bool,
};
