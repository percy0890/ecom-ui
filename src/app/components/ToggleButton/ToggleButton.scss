@import '~app/styles/core.scss';

.switch {
  position: relative;
  display: inline-block;
  width: 8rem;
  height: 4rem;

  input {
    opacity: 0;
    width: 0;
    height: 0;
  }

  .slider {
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: $app-text-tertiary;
    transition: 0.4s;
  }

  .slider:before {
    position: absolute;
    content: '';
    height: 3rem;
    width: 3rem;
    left: 1rem;
    bottom: 0.5rem;
    background-color: $app-text-white;
    transition: 0.4s;
  }

  input:checked + .slider {
    background-color: $app-logout-color;
  }

  input:focus + .slider {
    box-shadow: 0 0 1px $app-logout-color;
  }

  input:checked + .slider:before {
    transform: translateX(3rem);
  }

  /* Rounded sliders */
  .slider.round {
    border-radius: 2rem;
  }

  .slider.round:before {
    border-radius: 50%;
  }
}
