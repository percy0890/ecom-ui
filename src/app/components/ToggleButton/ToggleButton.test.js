import React from 'react';
import { shallow } from 'enzyme';
import ToggleButton from './ToggleButton';

describe('<ToggleButton />', () => {
  it('renders correctly', () => {
    expect(shallow(<ToggleButton buttonType="" />)).toMatchSnapshot();
  });
});
