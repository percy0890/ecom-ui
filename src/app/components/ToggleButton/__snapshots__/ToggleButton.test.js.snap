// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleButton /> renders correctly 1`] = `
<label
  className="switch "
>
  <input
    type="checkbox"
  />
  <span
    className="slider round"
  />
</label>
`;
